resources/url_shortener/responses/domain_response.yml

```diff
@@ -0,0 +1,11 @@
+type: object
+properties:
+  id:
+    description: Unique identifier for a domain.
+    type: string
+  domain:
+    description: The registered domain(hostname).
```

Comment:
You have `domain(hostname)` here and `domain/hostname` elsewhere. I'd stick with one or then update all occurrences to match.